boot/zephyr/main.c

```diff
@@ -313,6 +313,40 @@ void zephyr_boot_log_stop(void)
 #endif/* defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE) &&\
         !defined(CONFIG_LOG_PROCESS_THREAD) */
 
+#if defined(CONFIG_MCUBOOT_SERIAL) || defined(CONFIG_BOOT_USB_DFU_GPIO)
+static bool detect_pin(const char* port, int pin, uint32_t expected_value) {
+    int rc;
+    struct device const *detect_port;
+    uint32_t detect_value = !expected_value;
+
+    detect_port = device_get_binding(port);
+    __ASSERT(detect_port, "Error: Bad port for GPIO boot detection.\n");
+
+    /* The default presence value is 0 which would normally be
+     * active-low, but historically the raw value was checked so we'll
+     * use the raw interface.
+     */
+    rc = gpio_pin_configure(detect_port, pin,
+#ifdef GPIO_INPUT
+                            GPIO_INPUT | GPIO_PULL_UP
+#else
+                            GPIO_DIR_IN | GPIO_PUD_PULL_UP
+#endif
+           );
```

Comment:
I do not know whether mcuboot style allows that but it does not look nice.

Reply:
Essentially this entire function of code was extracted from below.

Reply:
My bad.